### PR TITLE
modified:   src/automerge.py

### PR DESCRIPTION
**Pull request by F-WRunTime**
- Simplified Check: The updated code retrieves the combined status of the latest commit directly.
- Single Condition: It checks if the combined status state is 'success' to determine if all checks have passed.
- Logging: It logs a single message if the checks are not passing, indicating the overall state.

### Benefits of the Change:

- Efficiency: The new approach reduces the complexity by avoiding the need to manually iterate over each required check.
- Readability: The code is more concise and easier to understand, focusing on the overall status rather than individual checks.
- Performance: By using the combined status, it potentially reduces the number of API calls and processing required.
- This change streamlines the process of determining if a PR is ready to be merged based on its status checks, making the code more maintainable and efficient.

Minimum requirements 
image
image
image
image

### CODE REVIEW

File: src/automerge.py [hunk 1]
```diff
@@ -90,21 +90,13 @@ def run_git_command(command_args: str) -> subprocess.CompletedProcess:
...
-            else:
-                check_run = latest_commit_checks[required_check]
-                if check_run.conclusion == 'success':
-                    print(f"Required check {required_check} passed on PR#{pr.number}")
-                else:
-                    print(f"Required check {required_check} failed or is pending on PR#{pr.number}")
-                    all_checks_passed = False
+        commit = pr.get_commits().reversed[0]
+        statuses = commit.get_statuses()
+        all_checks_passed = all(print(status) for status in statuses)
```
Comment (tothtamas28) [inline src/automerge.py]: This evaluates to `False` regardless of `statuses`. (And as a side effect, will print the first `status`.)

Comment (FWBlueWizard) [reply src/automerge.py]: The tests came back as expected. 
https://github.com/runtimeverification/automerge/actions/runs/12061856900/job/33634749102#step:4:1

I'll debug this now. Must be getting a false positive

Comment (FWBlueWizard) [reply src/automerge.py]: Addressed this issue. This required reworking permissions to introduce the use for a Github App. 
Remaining is to update the README and docs and tests

File: README.md [hunk 2]
```diff
@@ -112,8 +129,9 @@ Checkout the repository you wish to run automerge on to a local directory.
 git clone git@github.com:org/automerge.git
 cd automerge
 ```
-
-Now you need to run the command from this new directory 
+Setup GITHUB_TOKEN with the appropriate permissions. Content Read/Write, Pull Request Read/Write, adminstration Read, Checks Read.
```
Comment (tothtamas28) [inline README.md]:
```suggestion
Setup `GITHUB_TOKEN` with the appropriate permissions: Content Read/Write, Pull Request Read/Write, Adminstration Read, Checks Read. 
```

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**